Cabal/Distribution/Simple/Setup.hs

```diff
@@ -376,6 +376,7 @@ data ConfigFlags = ConfigFlags {
     configHcFlavor      :: Flag CompilerFlavor, -- ^The \"flavor\" of the
                                                 -- compiler, such as GHC or
                                                 -- JHC.
+    configTarget        :: Flag String,   -- ^the target prefix. (e.g. ghc -> TARGET-ghc)
```

Comment:
So, it's probably my fault that you edited this file, but given that the actual logic for configTarget lives solely in cabal-install, UNLESS you are planning on teaching Cabal the library how to interpret `--target` (you didn't mention anything about that) the flags don't belong here. Fortunately for you, cabal-install defines its own additional flags which you can pass on top; either `ConfigExFlags`/`BuildExFlags` or perhaps `InstallFlags` would be a better place to put them.